pkgs/os-specific/linux/kernel-headers/default.nix

```diff
@@ -93,6 +93,9 @@ in {
       };
       patches = [
          ./no-relocs.patch # for building x86 kernel headers on non-ELF platforms
+      ] ++ lib.optionals (stdenvNoCC.buildPlatform.isDarwin &&
```

Comment:
I think we generally try to have patches not be conditional.  Your patch already has `#ifdef APPLE` so it should be safe to apply it unconditionally, right?

Reply:
Since linuxHeaders is a critical package, and this cross scenario is such an edge case, I opted to patch conditionally so it incurs zero overhead for maintainers trying to bump the linux version. The patch may silently fail in the future, and the onus would be on me to notice this and update the patch. I see this done for other exotic scenarios, such as musl or BSD support, so it seemed fitting here.

Anyway, this is a moot point, because your other suggestion doesn't require patching and seems like a better solution overall :)